src/tools/dev/xml/GeneratePython.h

```diff
@@ -1200,15 +1200,15 @@ class AttsGeneratorUCharArray : public virtual UCharArray , public virtual Pytho
         c << "                else if(PyLong_Check(item))" << Endl;
         c << "                    c = int(PyLong_AsDouble(item));" << Endl;
         c << "                else" << Endl;
-        c << "                    c = 0;" << Endl;
+        c << "                    return PyExc_TypeError;" << Endl;
         c << Endl;
         c << "                if(c < 0) c = 0;" << Endl;
```

Comment:
We silently clamp `c` here to `[0...255]`. We could throw a `ValueError` instead.